src/app/tx/[tx]/MessageDisplays.tsx

```diff
@@ -0,0 +1,151 @@
+import dynamic from 'next/dynamic';
+import { ExternalLink } from '../../../components/ExternalLink';
+import {
+  getRetryableIdOrDepositHash,
+  l1ToL2MessageToStatusDisplay,
+} from '../../../lib';
+import {
+  L1ToL2MessagesAndDepositMessages,
+  MessageStatusDisplay,
+} from '../../../types';
+
+const getDataFromL1ToL2Message = (
+  l1ToL2Message: MessageStatusDisplay['l1ToL2Message'],
+) => {
+  if (l1ToL2Message && 'sender' in l1ToL2Message) {
+    const { sender, l1BaseFee, messageData } = l1ToL2Message;
+    const {
+      destAddress,
+      l2CallValue,
+      l1Value,
+      maxSubmissionFee,
+      excessFeeRefundAddress,
+      callValueRefundAddress,
+      gasLimit,
+      maxFeePerGas,
+      data,
+    } = messageData;
+    return {
+      sender,
+      l1BaseFee: l1BaseFee.toString(),
+      messageData: {
+        destAddress,
+        l2CallValue: l2CallValue.toString(),
+        l1Value: l1Value.toString(),
+        maxSubmissionFee: maxSubmissionFee.toString(),
+        excessFeeRefundAddress,
+        callValueRefundAddress,
+        gasLimit: gasLimit.toString(),
+        maxFeePerGas: maxFeePerGas.toString(),
+        data,
+      },
+    };
+  }
+
+  return {
+    sender: null,
+    l1BaseFee: null,
+    messageData: null,
+  };
+};
+
+async function getData({
+  retryables: l1ToL2Messages,
+  retryablesClassic: l1ToL2MessagesClassic,
+}: L1ToL2MessagesAndDepositMessages) {
+  const l1ToL2MessagesPromises = l1ToL2Messages.map((l1ToL2Message) =>
+    l1ToL2MessageToStatusDisplay(l1ToL2Message, false),
+  );
+  const l1ToL2MessagesClassicPromises = l1ToL2MessagesClassic.map(
+    (l1ToL2Message) => l1ToL2MessageToStatusDisplay(l1ToL2Message, true),
+  );
+
+  const messagesDisplays = await Promise.all(
+    l1ToL2MessagesPromises.concat(l1ToL2MessagesClassicPromises),
+  );
+
+  return { messagesDisplays };
+}
+
+type Props = {
+  messages: L1ToL2MessagesAndDepositMessages;
+  hasL2ToL1MessagesConfirmed: boolean;
+};
+
+const ConnectButtons = dynamic(
+  () => import('../../../components/ConnectButtons'),
+  {
+    ssr: false,
+  },
+);
+const Redeem = dynamic(() => import('../../../components/Redeem'), {
+  ssr: false,
+});
```

Comment:
We're using `ssr: false`, otherwise hydratation would be different on server and client